### PR TITLE
Feat/add item to list

### PR DESCRIPTION
**Pull request by nathanvandeperre97**

### CODE REVIEW

File: lib/pages/lists.dart [hunk 1]
```diff
@@ -5,6 +5,31 @@ class Lists extends StatelessWidget {
...
+          children: <Widget>[
+            Text(getString(context, "my_lists")),
+            RaisedButton(
+              onPressed: () => openOpenedListPage(context),
+              child: Text(
+                "open list",
+                style: TextStyle(
+                    color: Colors.grey[600], fontWeight: FontWeight.bold),
+              ),
+              color: Colors.white,
```
Comment (zohra051) [inline lib/pages/lists.dart]: please use const Color WHITE = Colors.white;

File: lib/pages/openedListPage.dart [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                child: FloatingActionButton(
+                  onPressed: () {
+                    showDialog(
+                      context: context,
+                      builder: (BuildContext context) =>
+                          _buildAddItemDialog(context),
+                    );
+                  },
+                  child: Icon(Icons.add),
+                  backgroundColor: Colors.green,
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use const Color GREEN = Colors.green;

File: lib/pages/openedListPage.dart [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+            crossAxisAlignment: CrossAxisAlignment.center,
+            mainAxisSize: MainAxisSize.min,
+            children: <Widget>[
+              TextField(
+                key: Key("item_name_label"),
+                textAlign: TextAlign.center,
+                decoration: InputDecoration(
+                  hintText: getString(context, 'item_name'),
+                  enabledBorder: OutlineInputBorder(
+                    borderSide: BorderSide(color: Colors.transparent),
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use const Color TRANSPARENT = Colors.transparent;

File: lib/pages/openedListPage.dart [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+                  ),
+                  filled: true,
+                  fillColor: Colors.transparent,
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use const Color TRANSPARENT = Colors.transparent;

File: lib/pages/openedListPage.dart [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: [
+                  Container(
+                    width: 35,
+                    child: FloatingActionButton(
+                      onPressed: () => decrementCounter(),
+                      child: Icon(Icons.remove),
+                      backgroundColor: Colors.red,
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: const Color RED = Colors.red;

File: lib/pages/openedListPage.dart [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+                    child: TextField(
+                      style: TextStyle(
+                        fontSize: 12,
+                      ),
+                      key: Key("item_count_label"),
+                      textAlign: TextAlign.center,
+                      decoration: InputDecoration(
+                        contentPadding: EdgeInsets.all(10),
+                        enabledBorder: OutlineInputBorder(
+                          borderSide: BorderSide(color: Colors.transparent),
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: change const color please

File: lib/pages/openedListPage.dart [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+                      onChanged: (text) =>
+                          handleSubmittedItemCount(int.parse(text)),
+                    ),
+                  ),
+                  Container(
+                    width: 35,
+                    child: FloatingActionButton(
+                      onPressed: () => incrementCounter(),
+                      child: Icon(Icons.add),
+                      backgroundColor: Colors.red,
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: change const color please

File: lib/pages/openedListPage.dart [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+                    ),
+                  ),
+                ],
+              ),
+              SizedBox(
+                height: 5,
+              ),
+              DropdownButtonFormField<String>(
+                hint: Text(getString(context, 'item_type')),
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use ),

File: lib/pages/openedListPage.dart [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+                height: 5,
+              ),
+              RaisedButton(
+                onPressed: () => addItemToList(),
+                child: Text(
+                  getString(context, 'add_item'),
+                  style: TextStyle(
+                      color: Colors.grey[600], fontWeight: FontWeight.bold),
+                ),
+                color: Colors.white,
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use const color

File: lib/pages/openedListPage.dart [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+              ),
+              SizedBox(
+                height: 5,
+              ),
+              RaisedButton(
+                onPressed: () => addItemToList(),
+                child: Text(
+                  getString(context, 'add_item'),
+                  style: TextStyle(
+                      color: Colors.grey[600], fontWeight: FontWeight.bold),
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use const color

File: lib/pages/openedListPage.dart [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+            child: Stack(
+              children: [
+                Positioned(
+                  top: 25,
+                  left: 10,
+                  child: Container(
+                    height: 50,
+                    width: 50,
+                    child: Image(
+                      image: AssetImage("assets/images/bottle_of_water.png"),
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: you can use Image("assets/images/bottle_of_water.png")

File: lib/pages/openedListPage.dart [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+                    ),
+                  ),
+                ),
+                Positioned(
+                  top: 60,
+                  left: 100,
+                  child: Text(
+                    itemCount,
+                    style: TextStyle(
+                      color: Colors.grey,
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use const color

File: lib/pages/openedListPage.dart [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+                    ),
+                  ),
+                ),
+                Positioned(
+                  left: 130,
+                  top: 60,
+                  child: Text(
+                    itemType,
+                    style: TextStyle(
+                      color: Colors.grey,
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: please use const color

File: lib/pages/openedListPage.dart [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+
+  void decrementCounter() {
+    if (_itemCount > 0) {
+      _itemCountController.text = (_itemCount - 1).toString();
+      _itemCount = _itemCount - 1;
+    }
+  }
+
+  void addItemToList() {
+    setState(() {
```
Comment (zohra051) [inline lib/pages/openedListPage.dart]: i think setState is not necessary ?

**Review by zohra051 — COMMENTED** [review]
please change

File: lib/pages/openedListPage.dart (hunk 14 above)
Comment (nathanvandeperre97) [reply lib/pages/openedListPage.dart]: If the setState is not used, we need to refresh the page to see the added item

File: lib/pages/openedListPage.dart (hunk 11 above)
Comment (nathanvandeperre97) [reply lib/pages/openedListPage.dart]: I can't use asset path with Image

File: lib/pages/openedListPage.dart (hunk 10 above)
Comment (nathanvandeperre97) [reply lib/pages/openedListPage.dart]: I can't use const with a Colors.color[xxx]

File: lib/pages/lists.dart [hunk 15]
```diff
@@ -5,6 +5,31 @@ class Lists extends StatelessWidget {
...
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            Text(getString(context, "my_lists")),
+            RaisedButton(
+              onPressed: () => openOpenedListPage(context),
+              child: Text(
+                "open list",
```
Comment (PiotrFLEURY) [inline lib/pages/lists.dart]: hard coded string

File: lib/pages/openedListPage.dart [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+
+  void handleSubmittedItemCount(int input) {
+    _itemCount = input;
+  }
+
+  void handleSubmittedItemName(String input) {
+    _itemName = input;
+  }
+
+  addDatabaseItemsToList() {}
```
Comment (PiotrFLEURY) [inline lib/pages/openedListPage.dart]: why this empty function ?

File: lib/pages/openedListPage.dart [hunk 17]
```diff
@@ -0,0 +1,310 @@
...
+  void initState() {
+    addDatabaseItemsToList();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      key: Key("openedList"),
+      resizeToAvoidBottomInset: false,
```
Comment (PiotrFLEURY) [inline lib/pages/openedListPage.dart]: what ?

File: lib/pages/openedListPage.dart [hunk 18]
```diff
@@ -0,0 +1,310 @@
...
+    if (_itemCount > 0) {
+      _itemCountController.text = (_itemCount - 1).toString();
+      _itemCount = _itemCount - 1;
+    }
+  }
+
+  void addItemToList() {
+    setState(() {
+      widgetList.add(
+          _addedItem(context, _itemName, _itemCount.toString(), _itemType));
```
Comment (PiotrFLEURY) [inline lib/pages/openedListPage.dart]: item should be an object instead of a widget

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

File: lib/pages/lists.dart (hunk 15 above)
Comment (nathanvandeperre97) [reply lib/pages/lists.dart]: This button will disappear on merge

File: lib/pages/openedListPage.dart (hunk 16 above)
Comment (nathanvandeperre97) [reply lib/pages/openedListPage.dart]: Because we don't get items from database yet

File: lib/pages/lists.dart [hunk 19]
```diff
@@ -44,8 +44,12 @@ class ListsState extends State<Lists> {
...
                   itemCount: wishlist.length,
                   itemBuilder: (BuildContext ctxt, int index) {
-                    return WidgetLists(wishlist[index],
-                        "assets/images/basket_my_lists.png", "0");
+                    return GestureDetector(
+                        onTap: () {
+                          openOpenedListPage(context, wishlist[index]);
+                        },
+                        child: WidgetLists(wishlist[index],
+                            "assets/images/basket_my_lists.png", "0 partage"));
```
Comment (PiotrFLEURY) [inline lib/pages/lists.dart]: 1) why are you giving the picture as parameter ? why don't you use the picture directly inside the WidgetLists widget ?

2) Hard coded string "0 partage" please use getString(...)

File: lib/pages/openedListPage.dart [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+      setState(() {
+        label = labelValue;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    listUuid = ModalRoute.of(context).settings.arguments;
+    getListTitle(listUuid);
```
Comment (PiotrFLEURY) [inline lib/pages/openedListPage.dart:50]: move getListTitle(...) call to iniState method

File: lib/pages/widget_list.dart [hunk 21]
```diff
@@ -5,6 +5,7 @@ import 'package:flutter/material.dart';
 const Color WHITE = Colors.white;
 const Color BLACK = Colors.black;
 const Color GREY = Colors.grey;
+String test = "";
```
Comment (PiotrFLEURY) [inline lib/pages/widget_list.dart]: what is this test var ?

File: lib/pages/widget_popup.dart [hunk 22]
```diff
@@ -0,0 +1,297 @@
...
+      _type = unitValue;
+      _name = labelValue;
+      _count = int.parse(quantityValue);
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    if (buttonName == "Update") {
```
Comment (PiotrFLEURY) [inline lib/pages/widget_popup.dart]: hard coded string "Update" please use getString()

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]
Please consider requested changes

File: lib/pages/lists.dart (hunk 19 above)
Comment (zohra051) [reply lib/pages/lists.dart]: because after we will change picture for each lists

Comment (zohra051) [reply lib/pages/lists.dart]: this string will change after

File: lib/pages/widget_popup.dart [hunk 23]
```diff
@@ -0,0 +1,301 @@
...
+    await Firestore.instance
+        .collection("items")
+        .document(listUuid)
+        .get()
+        .then((value) {
+      doesDocumentExist = value.exists;
+    });
+
+    //Create the item
+    if (doesDocumentExist == true) {
```
Comment (PiotrFLEURY) [inline lib/pages/widget_popup.dart:278]: doesDocumentExist will always be false. You call addItemToList only if you are not in update mode so it is not necessary to check if document exists. Just add it

File: lib/pages/widget_popup.dart [hunk 24]
```diff
@@ -0,0 +1,301 @@
...
+      });
+    } else {
+      await databaseReference.collection("items").document(listUuid).setData({
+        newUuid: {
+          'label': _name,
+          'quantity': _count,
+          'unit': _type,
+        }
+      });
+    }
```
Comment (PiotrFLEURY) [inline lib/pages/widget_popup.dart:297]: you forgot to increment the itemCount of the list

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

File: lib/pages/openedListPage.dart [hunk 25]
```diff
@@ -0,0 +1,135 @@
...
+
+const Color GREEN = Colors.green;
+const Color GREY = Colors.grey;
+const Color GREY600 = Colors.grey;
+const Color RED = Colors.red;
+const Color WHITE = Colors.white;
+const Color TRANSPARENT = Colors.transparent;
+
+final itemNameController = new TextEditingController();
+final itemCountController = new TextEditingController();
```
Comment (PiotrFLEURY) [inline lib/pages/openedListPage.dart]: why are you setting those two controllers in the OpenedListPage ? They are only used in the WidgetPopup. Please move thos two controllers in the WidgetPopup widget

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

File: lib/pages/widget_popup.dart [hunk 26]
```diff
@@ -0,0 +1,301 @@
...
+
+  WidgetPopupState(this.buttonName, this.listUuid, this.itemUuid);
+
+  String _name;
+  int _count;
+  String _type;
+
+  String alert = "";
+  String label = "";
+  String quantity = "";
```
Comment (PiotrFLEURY) [inline lib/pages/widget_popup.dart:42]: default quantity should be 1

File: lib/pages/widget_popup.dart [hunk 27]
```diff
@@ -0,0 +1,301 @@
...
+import 'package:flutter/services.dart';
+import 'package:uuid/uuid.dart';
+
+const Color WHITE = Colors.white;
+const Color BLACK = Colors.black;
+const Color GREY = Colors.grey;
+const Color RED = Colors.red;
+const Color TRANSPARENT = Colors.transparent;
+
+class WidgetPopup extends StatefulWidget {
```
Comment (PiotrFLEURY) [inline lib/pages/widget_popup.dart]: Wrong name. Please give it an explicit name as EditItemPopup

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

**Review by PiotrFLEURY — APPROVED** [review]

**Pull request merged**